### PR TITLE
Restricting cornac to 1.15.1 for issue with 1.15.4

### PR DESCRIPTION
**Pull request by miguelgfierro**
### Description

See https://github.com/microsoft/recommenders/issues/1933

### Checklist:

- [ ] I have followed the [contribution guidelines](../CONTRIBUTING.md) and code style for this project.
- [ ] I have added tests covering my contributions.
- [ ] I have updated the documentation accordingly.
- [ ] This PR is being made to `staging branch` and not to `main branch`.

### CODE REVIEW

**Comment by miguelgfierro** [comment]
mmmm it seems it is not taking the changing, it still uses cornac 1.15.4, instead of 1.15.1. 



```
Collecting cornac<2,>=1.1.2
  Downloading cornac-1.15.4.tar.gz (7.9 MB)
     â”�â”�â”�â”�â”�â”�â”�â”�â”�â”�â”�â”�â”�â”�â”�â”�â”�â”�â”�â”�â”�â”�â”�â”�â”�â”�â”�â”�â”�â”�â”�â”�â”�â”�â”�â”�â”�â”�â”�â”� 7.9/7.9 MB 121.1 MB/s eta 0:00:00
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'error'

failed
Pip subprocess error:
  error: subprocess-exited-with-error
  
  Ã— python setup.py egg_info did not run successfully.
  â”‚ exit code: 1
  â•°â”€> [6 lines of output]
      Traceback (most recent call last):
        File "<string>", line 36, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/tmp/pip-install-pfz_52us/cornac_4b74720c302545d1abaa7375b4556cb1/setup.py", line 32, in <module>
          import numpy as np
      ModuleNotFoundError: No module named 'numpy'
      [end of output]
```

Why don't you obey me?

o_o

**Comment by miguelgfierro** [comment]
```
Installing pip dependencies: ...working... Ran pip subprocess with arguments:
['/azureml-envs/azureml_8fe95e22b208ec42145257f7dcae20e3/bin/python', '-m', 'pip', 'install', '-U', '-r', '/azureml-environment-setup/condaenv.xes2msz1.requirements.txt', '--exists-action=b']
Pip subprocess output:

failed
Pip subprocess error:
ERROR: Directory '.[dev,examples,gpu]' is not installable. Neither 'setup.py' nor 'pyproject.toml' found.
```

**Comment by miguelgfierro** [comment]
@pradnyeshjoshi I'm trying to install the local recommenders `pip install .` instead of the pip version (see messages above) using RunConfiguration() class of AzureML. But I can't find a solution. Do you have experience with it?

**Comment by miguelgfierro** [comment]
tried with command:
```
    # install extra dependencies
    if add_gpu_dependencies and add_spark_dependencies:
        conda_dep.add_channel("conda-forge")
        conda_dep.add_conda_package(conda_pkg_jdk)
        # conda_dep.add_pip_package("recommenders[dev,examples,spark,gpu]") # Not working, it installs the pip package instead of the local wheel
        # run_azuremlcompute.script = ["pip", "install", ".[dev,examples,spark,gpu]"] # not working
        run_azuremlcompute.command = ["pip", "install", ".[dev,examples,spark,gpu]"] 
    elif add_gpu_dependencies:
        # conda_dep.add_pip_package("recommenders[dev,examples,gpu]") # Not working, it installs the pip package instead of the local wheel
        # run_azuremlcompute.script = ["pip", "install", ".[dev,examples,gpu]"] # not working
        run_azuremlcompute.command = ["pip", "install", ".[dev,examples,gpu]"]
    elif add_spark_dependencies:
        conda_dep.add_channel("conda-forge")
        conda_dep.add_conda_package(conda_pkg_jdk)
        # conda_dep.add_pip_package("recommenders[dev,examples,spark]") # Not working, it installs the pip package instead of the local wheel
        # run_azuremlcompute.script = ["pip", "install", ".[dev,examples,spark]"] # not working
        run_azuremlcompute.command = ["pip", "install", ".[dev,examples,spark]"]

    else:
        # conda_dep.add_pip_package("[dev,examples]")
        # run_azuremlcompute.script = ["pip", "install", ".[dev,examples]"]
        run_azuremlcompute.command = ["pip", "install", ".[dev,examples]"]
```
I got the error: Both Script and Command cannot be specified with the request

**Comment by miguelgfierro** [comment]
Tried interpreter options:
```
    # install extra dependencies
    if add_gpu_dependencies and add_spark_dependencies:
        conda_dep.add_channel("conda-forge")
        conda_dep.add_conda_package(conda_pkg_jdk)
        # conda_dep.add_pip_package("recommenders[dev,examples,spark,gpu]") # Not working, it installs the pip package instead of the local wheel
        # run_azuremlcompute.script = ["pip", "install", ".[dev,examples,spark,gpu]"] # not working
        # run_azuremlcompute.command = ["pip", "install", ".[dev,examples,spark,gpu]"]  # Error: Both Script and Command cannot be specified with the request
        run_azuremlcompute.environment.python.interpreter_options = "-m venv .venv && source .venv/bin/activate && pip install -e .[dev,examples,spark,gpu]"    
    elif add_gpu_dependencies:
        # conda_dep.add_pip_package("recommenders[dev,examples,gpu]") # Not working, it installs the pip package instead of the local wheel
        # run_azuremlcompute.script = ["pip", "install", ".[dev,examples,gpu]"] # not working
        # run_azuremlcompute.command = ["pip", "install", ".[dev,examples,gpu]"] # Error: Both Script and Command cannot be specified with the request
        run_azuremlcompute.environment.python.interpreter_options = "-m venv .venv && source .venv/bin/activate && pip install -e .[dev,examples,gpu]"
    elif add_spark_dependencies:
        conda_dep.add_channel("conda-forge")
        conda_dep.add_conda_package(conda_pkg_jdk)
        # conda_dep.add_pip_package("recommenders[dev,examples,spark]") # Not working, it installs the pip package instead of the local wheel
        # run_azuremlcompute.script = ["pip", "install", ".[dev,examples,spark]"] # not working
        # run_azuremlcompute.command = ["pip", "install", ".[dev,examples,spark]"] # Error: Both Script and Command cannot be specified with the request
        run_azuremlcompute.environment.python.interpreter_options = "-m venv .venv && source .venv/bin/activate && pip install -e .[dev,examples,spark]"
    else:
        # conda_dep.add_pip_package("[dev,examples]")
        # run_azuremlcompute.script = ["pip", "install", ".[dev,examples]"] # not working
        # run_azuremlcompute.command = ["pip", "install", ".[dev,examples]"] # Error: Both Script and Command cannot be specified with the request
        run_azuremlcompute.environment.python.interpreter_options = "-m venv .venv && source .venv/bin/activate && pip install -e .[dev,examples]"

```
got the error: `AttributeError: <class 'azureml.core.environment.PythonSection'> has no attribute interpreter_options`

File: tests/ci/azureml_tests/submit_groupwise_azureml_pytest.py [hunk 1]
```diff
@@ -179,43 +179,83 @@ def create_run_config(
...
-        conda_dep.add_channel("conda-forge")
-        conda_dep.add_conda_package(conda_pkg_jdk)
-        conda_dep.add_pip_package("recommenders[dev,examples,spark]")
-    else:
-        conda_dep.add_pip_package("recommenders[dev,examples]")
+    # Enable DockerSection https://learn.microsoft.com/en-us/python/api/azureml-core/azureml.core.environment.dockersection
+    run_azuremlcompute.environment.docker.enabled = True
+    # run_azuremlcompute.environment.docker.base_image = docker_proc_type
+    
+    dockerfile = """
```
Comment (simonzhaoms) [inline tests/ci/azureml_tests/submit_groupwise_azureml_pytest.py:187]: I think it is more reliable to use `CondaDependencies` to set up the Conda env than dockerfile

Comment (simonzhaoms) [reply tests/ci/azureml_tests/submit_groupwise_azureml_pytest.py:187]: @miguelgfierro I see why you tried to use dockerfile here.  I think it should be possible to use `CondaDependencies` to install recommenders locally.  I'll do a research.

**Comment by miguelgfierro** [comment]
This is awesome @simonzhaoms

**Pull request merged**